app/scripts/translator-screenshots/lib.js

```diff
@@ -0,0 +1,473 @@
+#!/usr/bin/env node
+
+import { _electron as electron } from "playwright";
+import { existsSync, mkdirSync } from "fs";
+import { join } from "path";
+import TranslationExtractor from "./extractor.js";
+
+/**
+ * Clean API for automated translation screenshots
+ */
+class TranslationScreenshotAPI {
+  constructor(options = {}) {
+    this.app = null;
+    this.page = null;
+    this.outputDir = options.outputDir || "./screenshots";
+    this.currentContext = null;
+    this.extractor = new TranslationExtractor();
+    this.screenshotState = new Set(); // Track taken screenshots
+    this.allAttemptedKeys = new Set(); // Track all keys we've tried
+
+    // Ensure output directory exists
+    if (!existsSync(this.outputDir)) {
+      mkdirSync(this.outputDir, { recursive: true });
+    }
+  }
+
+  /**
+   * Initialize and launch the Electron app
+   */
+  async launch() {
+    console.log("Launching Electron app...");
+
+    const args = ["./out/main/index.js"];
+    const isCI = Boolean(process.env.CI);
+
+    if (isCI) {
+      args.push("--no-sandbox");
+      console.log("Running in CI mode, disabling sandbox");
+    }
+
+    this.app = await electron.launch({
+      args: args,
+      headless: isCI,
+    });
+
+    this.page = await this.app.firstWindow();
+    await this.page.setViewportSize({ width: 1920, height: 1080 });
+    await this.page.waitForLoadState("networkidle");
+
+    console.log("App launched at 1920x1080");
+  }
+
+  /**
+   * Set the current translation context/namespace
+   * @param {string} namespace - Translation namespace (e.g., "SignIn", "MainContent")
+   */
+  context(namespace) {
+    this.currentContext = namespace;
+    console.log(`Set context to: ${namespace}`);
+  }
+
+  /**
+   * Navigate to a specific route
+   * @param {string} route - Route to navigate to (e.g., "#/signin")
+   */
+  async navigate(route) {
+    console.log(`Navigating to: ${route}`);
+
+    if (route.startsWith("#")) {
+      // Client-side routing
+      const currentUrl = this.page.url();
+      const baseUrl = currentUrl.split("#")[0];
+      await this.page.goto(`${baseUrl}${route}`);
+    } else {
+      await this.page.goto(route);
+    }
+
+    await this.page.waitForLoadState("networkidle");
+  }
+
+  /**
+   * Click an element by test ID
+   * @param {string} testId - data-testid attribute value
+   */
+  async clickByTestId(testId) {
+    console.log(`Clicking element with testid: ${testId}`);
+    const element = this.page.getByTestId(testId);
+    await element.click();
+    await this.page.waitForLoadState("networkidle");
+  }
+
+  /**
+   * Hover over an element by test ID (useful for tooltips)
+   * @param {string} testId - data-testid attribute value, use * for wildcard suffix
+   */
+  async hoverByTestId(testId) {
```

Comment:
can we consolidate `hoverByTestId` and the `getByTestId().setValue/click` interfaces?

the wildcard matching we have here seems useful, so go support wildcard matching in `getByTestId()` and then we could have `hover()` be a method within there (like with `setValue/click`)

Reply:
oh and the same for `clickByTestId` as well

Reply:
admittedly the wildcard matching mostly exists because the UUIDs change. if we were using a fixed dataset (#2720), then we could hardcode the UUID and wouldn't really need it 🤔

Reply:
that's fair! I'm good with removing the wildcard matching, and more so feel it would be nice to have either `hoverByTestId`/`clickByTestId` _or_ `getByTestId().hover()` and `getByTestId().click()` as the paths rather than two ways of doing the same action